### PR TITLE
Use modern rules_python and expect users to provide the mypy library themselves

### PR DESCRIPTION
**Pull request by martis42**
New version of https://github.com/bazel-contrib/bazel-mypy-integration/pull/82

As requested, rules_python is now the latest version still supporting Bazel 5.

There is still no default version for `mypy`, as the discussion about this is still ongoing.
I still believe there should be no default version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,35 +72,24 @@ If there's a typing error in your Python code, then the test will fail. Using `-
 
 ## Installation
 
-**1. Create a file (`tools/typing/mypy_version.txt` in the example below) that will specify the version of `mypy` to use.** You will pass the Bazel label for
-this file to the `deps()` function in `@mypy_integration//repositories:deps.bzl`, which below is named
-`mypy_integration_deps(...)`:
+`mypy_integration` expects the user to provide the `mypy` dependency.
+Given not every mypy version is compatible to all Python versions and mypy's transitive dependencies can differ based on the Python version `mypy_integration` cannot offer a `mypy` which satisfies all potential users.
```
Comment (adzenith) [inline README.md:76]:
```suggestion
Given not every mypy version is compatible to all Python versions and mypy's transitive dependencies can differ based on the Python version, `mypy_integration` cannot offer a `mypy` which satisfies all potential users.
```

**Review by adzenith — APPROVED** [review]
Looks great! Thank you!

File: README.md (hunk 1 above)
Comment (martis42) [reply README.md:76]: Will fix this in another PR so this large PR can be merged without a new review.

Comment (adzenith) [reply README.md:76]: Ah right sorry, at work when I approve a PR the author hits merge when they're ready, so here out of habit I approved and then didn't hit merge. I'm happy to merge whenever you're ready.

**Comment by adzenith** [comment]
thank you!!

**Pull request merged**